cmd/context.go

```diff
@@ -0,0 +1,39 @@
+package cmd
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/charmbracelet/log"
+)
+
+var onlyOneSignalHandler = make(chan struct{}) //nolint: gochecknoglobals
+
+// SignalHandledContext returns a context that will be canceled if a SIGINT or SIGTERM is
+// received.
+func SignalHandledContext() (context.Context, context.CancelFunc) {
+	// panics when called twice, this way there can only be one signal handled context
+	close(onlyOneSignalHandler)
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	sigs := make(chan os.Signal, 2) //nolint:mnd
+
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+
+	go func() {
+		sig := <-sigs
+		log.Infof("received signal '%s', canceling context", sig)
+
+		cancel()
+
+		sig = <-sigs
+		log.Infof("received signal '%s', exiting program", sig)
+
+		os.Exit(1)
+	}()
```

Comment:
it seems that the logic is not the same with the new ctrlc handler

previously, when the context was cancelled, the destroy op was kicked in to remove the partially deployed lab and leave no traces:

```go
		// when interrupted, destroy the interrupted lab deployment
		cleanup = false

		if err := destroyFn(destroyCmd, []string{}); err != nil {
			log.Errorf("Failed to destroy lab: %v", err)
		}

		os.Exit(1) // skipcq: RVV-A0003
```